### PR TITLE
Implement Remove/Add Functionality

### PR DESCRIPTION
**Pull request by mohamedSabry0**
Implement the following methods:
- Remove task.
- Add task.
- Remove all completed.
- Edit task.

### CODE REVIEW

File: src/remove_all_completed.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import storage from './store.js';
+
+const clearAllCompleted = () => {
+  let list = storage.get();
+  list = list.filter((task) => !task.completed);
+  storage.setList(list);
+};
```
Comment (sinansevgi) [inline src/remove_all_completed.js:7]: - [ ] `clearAllCompleted` method doesn't update indexes of the items after removing completed tasks from the list.
And if I try to add a new item to the list afterward it causes duplicate indexes.
Screencast from 08-29-2021 12_22_08 AM

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @mohamedSabry0 , 

Great Work :+1: 

Good job using array methods like map, filter and forEach :clap:
There are some issues that you still need to work on to go further but you are almost there!


 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by de-scholar — APPROVED** [review]
Hi @mohamedSabry0 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- [ ] [Optional] Just an optional requirement: you can try to align the checkbox with the text, for a better UI/UX

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/remove_all_completed.js (hunk 1 above)
Comment (de-scholar) [reply src/remove_all_completed.js]: :+1: This issue is resolved :heavy_check_mark:

**Pull request merged**